python/screen_grab/screen_grab.py

```diff
@@ -0,0 +1,211 @@
+                            QtCore.Qt.WindowStaysOnTopHint |
+                            QtCore.Qt.CustomizeWindowHint |
+                            QtCore.Qt.Tool)
+        self.setAttribute(QtCore.Qt.WA_TranslucentBackground)
+        self.setCursor(QtCore.Qt.CrossCursor)
+        self.setMouseTracking(True)
+
+        desktop = QtGui.QApplication.instance().desktop()
+        desktop.resized.connect(self._fit_screen_geometry)
+        desktop.screenCountChanged.connect(self._fit_screen_geometry)
+
+    @property
+    def capture_rect(self):
+        """
+        The resulting QRect from a previous capture operation.
+        """
+        return self._capture_rect
+
+    def paintEvent(self, evt):
+        """
+        Paint event
+        """
+        # Convert click and current mouse positions to local space.
+        mouse_pos = self.mapFromGlobal(QtGui.QCursor.pos())
+        click_pos = None
+        if self._click_pos is not None:
+            click_pos = self.mapFromGlobal(self._click_pos)
+
+        painter = QtGui.QPainter(self)
+
+        # Draw background. Aside from aesthetics, this makes the full
+        # tool region accept mouse events.
+        painter.setBrush(QtGui.QColor(0, 0, 0, self._opacity))
+        painter.setPen(QtCore.Qt.NoPen)
+        painter.drawRect(evt.rect())
+
+        # Clear the capture area
+        if click_pos is not None:
+            capture_rect = QtCore.QRect(click_pos, mouse_pos)
+            painter.setCompositionMode(QtGui.QPainter.CompositionMode_Clear)
+            painter.drawRect(capture_rect)
+            painter.setCompositionMode(QtGui.QPainter.CompositionMode_SourceOver)
+
+        pen = QtGui.QPen(QtGui.QColor(255, 255, 255, 64), 1, QtCore.Qt.DotLine)
+        painter.setPen(pen)
+
+        # Draw cropping markers at click position
+        if click_pos is not None:
+            painter.drawLine(evt.rect().left(), click_pos.y(),
+                             evt.rect().right(), click_pos.y())
+            painter.drawLine(click_pos.x(), evt.rect().top(),
+                             click_pos.x(), evt.rect().bottom())
+
+        # Draw cropping markers at current mouse position
+        painter.drawLine(evt.rect().left(), mouse_pos.y(),
+                         evt.rect().right(), mouse_pos.y())
+        painter.drawLine(mouse_pos.x(), evt.rect().top(),
+                         mouse_pos.x(), evt.rect().bottom())
+
+    def keyPressEvent(self, evt):
+        """
+        Key press event
+        """
+        # for some reason I am not totally sure about, it looks like
+        # pressing escape while this dialog is active crashes Maya.
+        # I tried subclassing closeEvent, but it looks like the crashing
+        # is triggered before the code reaches this point.
+        # by sealing the keypress event and not allowing any further processing
+        # of the escape key (or any other key for that matter), the 
+        # behaviour can be successfully avoided.
```

Comment:
I tried doing this, but wasn't successful - for some reason it crashes in Maya. I didn't have time to investigate so I just "plugged the whole" with the fix above. Admittedly, I didn't dig very deep - just trying some obvious stuff - and it would be nice to be able to hit ESC to cancel - that's how the macosx screen grabber work for example. At the same time, it feels like a nice to have! :) Unless you object, I suggest put a ticket in so we can potentially address it later. What do you think?